modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -390,6 +390,12 @@ def to_numpy(self, **kwargs):
     combine = Binary.register(pandas.DataFrame.combine, infer_dtypes="common_cast")
     combine_first = Binary.register(pandas.DataFrame.combine_first, infer_dtypes="bool")
     eq = Binary.register(pandas.DataFrame.eq, infer_dtypes="bool")
+    equals = Binary.register(
+        lambda df, other: pandas.DataFrame([df.equals(other)]),
+        join_type=None,
```

Comment:
Indexes are checked at a higher level, so there is no need to do a join, it is known for sure that the axes are equal.